apps/metric_processors.go

```diff
@@ -42,7 +41,9 @@ func (p MetricsProcessorExcludeMetrics) Processors() []otel.Component {
 		for _, g := range strings.Split(glob, "*") {
 			literals = append(literals, regexp.QuoteMeta(g))
 		}
-		metricNames = append(metricNames, fmt.Sprintf(`^%s$`, strings.Join(literals, `.*`)))
+		// All filter regexes are automatically anchored.
+		// https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/processor/filterset/regexp/regexpfilterset.go#L79
```

Comment:
It's not anchored. Though it did solve the problem.

Reply:
Discussed offline. @sophieyfang found the actual cause of this, which was #151 not updating [the removal of the prefix](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/apps/metric_processors.go#L40). I'll update this PR and request another review after testing.